tests/test_observers/test_file_storage_observer.py

```diff
@@ -117,11 +117,21 @@ def test_fs_observer_started_event_creates_rundir(dir_obs, sample_run, monkeypat
     def mkdir_raises_file_exists(name, mode=0o777):
         raise FileExistsError("File already exists: " + name)
 
+    # Assume some problem with the filesystem exists
+    #     therefore run dir creation should stop after some tries
     with monkeypatch.context() as m:
         m.setattr('os.mkdir', mkdir_raises_file_exists)
         with pytest.raises(FileExistsError):
             sample_run['_id'] = None
-            _id = obs.started_event(**sample_run)
+            obs.started_event(**sample_run)
+
+    # Assume listdir doesn't show existing file (e.g. due to caching or delay of network storage)
+    assert os.listdir(str(basedir)) == [_id]
+    with monkeypatch.context() as m:
+        m.setattr('os.listdir', lambda __: [])
```

Comment:
Nitpicking, but I would use only a single underscore for the irrelevant lambda argument.

Reply:
Sure, such it is more consistent in the codebase.

Just an explanation why i used it: It is prefered by several people to use double-underline, because single-underline is heavily over-used. E.g. REPL sets the last output as _.[gettext.gettext](https://docs.python.org/3/library/gettext.html) usually aliased as _.